linak-desk-control.py

```diff
@@ -172,6 +172,10 @@ def __init__(self, vendor_id=0x12d3, product_id=0x0002):
 		if not self._handle:
 			raise Exception('Could not connect to usb device')
 
+		if self._handle.kernelDriverActive(0):
```

Comment:
Does not work on Windows. Proposal to limit it to Linux ("posix").
Requires in addition to add an `import os` at beginning of file.

```suggestion
		if os.name == 'posix' and self._handle.kernelDriverActive(0):
```

Not sure whether this works on MacOS.

Reply:
Alternative: Catch `usb1.USBErrorNotSupported`

Reply:
I think the OS posix check is the way to go. I don't think we should catch USB errors down in the lib, as a failure to detach the kernel driver for any reason likely means the application will not work.

Reply:
Thank you @wormoworm !

Reply:
No worries - thanks for the work you've done in getting the project this far! Would you be OK if I use parts of this (credited of course) in a project of my own? I'm working on controlling a CBD6S over MQTT.

Reply:
Sure, feel free.